### DISCUSSION TITLE
CaseKeyPath from a CasePathable instance

### DISCUSSION THREAD

**Discussion by Ryu0118** [pointfreeco/swift-case-paths#136] [Ideas]
Hi, 
I would like to suggest an enhancement for enums conforming to CasePathable by providing a means to obtain a CaseKeyPath from an instance. Currently, to determine if two instances of CasePathable are the same case, we need to write code like the following:

```Swift
@CasePathable
enum Foo {
    case c1
    case c2(Int)

    var caseKeyPath: PartialCaseKeyPath<Self> {
        switch self {
        case .c1:
            return \.c1
        case .c2:
            return \.c2
        }
    }
}

let a: Foo = .c2(0)
let b: Foo = .c2(1)

print(a.is(b.caseKeyPath)) // true
```
Writing this code repeatedly is quite cumbersome. Therefore, I propose that the caseKeyPath should be generated by the CasePathable macro, to simplify the process and improve developer experience.

**Comment by stephencelis** [comment]
@Ryu0118 Thanks for the discussion. It's come up a few times :) We decided to implement via #158, which should hopefully be released soon. Let us know if you have any comments!
